x/ccv/provider/keeper/migration.go

```diff
@@ -0,0 +1,183 @@
+package keeper
+
+import (
+	"time"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	paramtypes "github.com/cosmos/cosmos-sdk/x/params/types"
+	ibctmtypes "github.com/cosmos/ibc-go/v4/modules/light-clients/07-tendermint/types"
+	providertypes "github.com/cosmos/interchain-security/v2/x/ccv/provider/types"
+	ccvtypes "github.com/cosmos/interchain-security/v2/x/ccv/types"
+)
+
+// Migrator is a struct for handling in-place store migrations.
+type Migrator struct {
+	ccvProviderKeeper     Keeper
+	ccvProviderParamSpace paramtypes.Subspace
+}
+
+// NewMigrator returns a new Migrator.
+func NewMigrator(ccvProviderKeeper Keeper, ccvProviderParamSpace paramtypes.Subspace,
+) Migrator {
+	return Migrator{ccvProviderKeeper: ccvProviderKeeper, ccvProviderParamSpace: ccvProviderParamSpace}
+}
+
+// Migratev1p1p0MultidenTov2 migrates a provider from v1.1.0-multiden to v2.0.0.
+func (m Migrator) Migratev1p1p0MultidenTov2(ctx sdk.Context) error {
+	return m.Migratev1Tov2(ctx) // Note same as v1.0.0 to v2.0.0
+}
+
+// Migratev1Tov2 migrates a provider from v1.0.0 to v2.0.0.
+func (m Migrator) Migratev1Tov2(ctx sdk.Context) error {
+	// Migrate params
+	MigrateParamsv1Tov2(ctx,
+		m.ccvProviderParamSpace,
+		// See https://github.com/cosmos/interchain-security/blob/7861804cb311507ec6aebebbfad60ea42eb8ed4b/x/ccv/provider/keeper/params.go#L84
+		// The v1.1.0-multiden version of ICS hardcodes this param as 10 of bond type: k.stakingKeeper.BondDenom(ctx).
+		// Here we use the same starting value, but the param can now be changed through governance.
+		sdk.NewCoin(m.ccvProviderKeeper.BondDenom(ctx), sdk.NewInt(10000000)),
```

Comment:
Perhaps `consumerRewardDenomRegistrationFee` can be just declared as a global variable.

Reply:
Is there a way to do this while also being able to access `m.ccvProviderKeeper.BondDenom(ctx)`?

Reply:
Good point, don't think so. U can withdraw this.